File: src/types/conversation.ts
/**
 * Conversation interface
 * Represents a saved conversation from an AI platform
 */
export interface Conversation {
  id?: number;
  canonical_url: string; // Normalized URL used for deduplication
  share_url?: string;
  source: 'chatgpt' | 'claude' | 'perplexity' | 'kimi' | 'mistral' | 'deepseek' | 'qwen' | 'manus' | 'grok' | 'other';
  title: string;
  description?: string;
  content: string; // Full content in markdown format
  tags: string[];
  collection_id?: number;
  ignore: boolean; // If true, Beast Mode will not collect for this URL
  version: number; // Incremented on each Beast Mode overwrite
  created_at: Date;
  updated_at: Date;
}

/**
 * Conversation payload for extraction
 * Used when extracting from content scripts
 */
export interface ConversationPayload {
  title: string;
  content: string; // Markdown formatted
  description?: string;
  shareUrl?: string;
}

File: server/application/internal/models/conversation.go
package models

import "time"

// Conversation represents a saved conversation from an AI platform
type Conversation struct {
	ID             *int       `json:"id,omitempty" db:"id"`
	CanonicalURL   string     `json:"canonical_url" db:"canonical_url"`
	ShareURL       *string    `json:"share_url,omitempty" db:"share_url"`
	Source         string     `json:"source" db:"source"`
	Title          string     `json:"title" db:"title"`
	Description    *string    `json:"description,omitempty" db:"description"`
	Content        string     `json:"content" db:"content"`
	Tags           []string   `json:"tags" db:"tags"`
	CollectionID   *int       `json:"collection_id,omitempty" db:"collection_id"`
	Ignore         bool       `json:"ignore" db:"ignore"`
	Version        int        `json:"version" db:"version"`
	CreatedAt      time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt      time.Time  `json:"updated_at" db:"updated_at"`
}

